### ISSUE TITLE
`vue/require-default-prop` doesn't trigger with props destructure

### ISSUE THREAD

**Issue by hartbit** [vuejs/eslint-plugin-vue#2725]
**Checklist**

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 9.21.0
- **eslint-plugin-vue version:** 10.0.0
- **Vue version:** 3.5.13
- **Node version:** 22.14.0
- **Operating System:** macOS

**Please show your full configuration:**
```ts
import pluginVue from "eslint-plugin-vue";
import { defineConfigWithVueTs, vueTsConfigs } from "@vue/eslint-config-typescript";

export default defineConfigWithVueTs(
  {
    name: "app/files-to-lint",
    files: ["**/*.{ts,mts,tsx,vue}"],
  },

  {
    name: "app/files-to-ignore",
    ignores: ["**/dist/**", "**/dist-ssr/**", "**/coverage/**"],
  },

  pluginVue.configs["flat/recommended"],
  vueTsConfigs.recommended
);

```

**What did you do?**	
```vue
const { msg } = defineProps<{
  msg?: string;
}>();
```

**What did you expect to happen?**
I switched from using the `withDefaults` to the new [props destructure syntax](https://vuejs.org/api/sfc-script-setup.html#reactive-props-destructure) and I expected the `vue/require-default-prop` to continue to check when defaults were not defined in the destructured object. I expected the above example to complain that I did not define defaults for `msg`.

**What actually happened?**
I got no errors in the above example.

**Repository to reproduce this issue**
https://github.com/hartbit/eslint-require-default-prop-destructured-issue
* The `src/components/HelloWorld.vue` component generates a warning, as expected, as it uses `withDefaults`,
* The `src/components/HelloWordIssue.vue` component does not generate a warning

**Comment by ZexTech** [comment]
The potentially unwanted change with this rule is that it now requires props to be defined (with undefined value).
This results in unnecessarily verbose code, despite the types being already correctly defined

```ts
// Before ✅
const { optional } = defineProps<{
  optional?: boolean;
  optionalLabel?: string;
}>();

// After ❌
const { optional, optionalLabel = undefined } = defineProps<{
  optional?: boolean;
  optionalLabel?: string;
}>();
```

Proposed solution could be to set this rule to off by default, minimizing the impact on users during the update to this version. This change would primarily affect users of the latest Vue 3.5 version who are utilizing destructured props.

Alternatively, a lint rule option could be introduced to require only that a default value is set, provided the property is explicitly added. 
``` const { [key] } = defineProps<... ```

**Comment by GrantGryczan** [comment]
Can the fix for this please be reverted? This issue should be considered working as intended. I explain why in https://github.com/vuejs/eslint-plugin-vue/issues/2741#issuecomment-3383619425. For months, I've been running into needless warnings that I have to `@eslint-disable-next-line` every time I write an optional prop. I'm trying to avoid disabling any default lints, because I strive to write code that's considered idiomatic for the framework I'm writing in. But this issue misses the lint's purpose, and its fix forces people to write code that isn't idiomatic.

EDIT: I believe I was misunderstanding the purpose of the lint. See https://github.com/vuejs/eslint-plugin-vue/issues/2741#issuecomment-3685617803 and my following reply. In particular, I think this fix is fine, but this rule should probably be disabled by default.
